ucte/ucte-converter/src/main/java/com/powsybl/ucte/converter/UcteExporter.java

```diff
@@ -0,0 +1,1072 @@
+        double distance = Math.sqrt(Math.pow(xb - xa, 2) + Math.pow(yb - ya, 2));
+        double du = 100 * distance / (tapNumber - 1);
+        return BigDecimal.valueOf(du).setScale(4, RoundingMode.HALF_UP).doubleValue();
+    }
+
+    /**
+     * Calculate the Θ for the angle regulation of the two windings transformer
+     *
+     * @param twoWindingsTransformer The twoWindingsTransformer containing the PhaseTapChanger we want to convert
+     * @return the Θ needed to create a {@link UcteAngleRegulation}
+     */
+    private double calculateAsymmAngleTheta(TwoWindingsTransformer twoWindingsTransformer) {
+        PhaseTapChanger phaseTapChanger = twoWindingsTransformer.getPhaseTapChanger();
+        int lowTapPosition = phaseTapChanger.getLowTapPosition();
+        int highTapPosition = phaseTapChanger.getHighTapPosition();
+        double lowPositionAlpha = Math.toRadians(-phaseTapChanger.getStep(lowTapPosition).getAlpha());
+        double lowPositionRho = 1 / phaseTapChanger.getStep(lowTapPosition).getRho();
+        double highPositionAlpha = Math.toRadians(-phaseTapChanger.getStep(highTapPosition).getAlpha());
+        double highPositionRho = 1 / phaseTapChanger.getStep(highTapPosition).getRho();
+        double xa = lowPositionRho * Math.cos(lowPositionAlpha);
+        double ya = lowPositionRho * Math.sin(lowPositionAlpha);
+        double xb = highPositionRho * Math.cos(highPositionAlpha);
+        double yb = highPositionRho * Math.sin(highPositionAlpha);
+        return Math.toDegrees(Math.atan((yb - ya) / (xb - xa)));
+    }
+
+    /**
+     * @param twoWindingsTransformer The twoWindingsTransformer containing the PhaseTapChanger we want to convert
+     * @return P (MW) of the angle regulation for the two windings transformer
+     */
+    private float calculateAngleP(TwoWindingsTransformer twoWindingsTransformer) {
+        return (float) twoWindingsTransformer.getPhaseTapChanger().getRegulationValue();
+    }
+
+    /**
+     * Give the type of the UcteAngleRegulation
+     *
+     * @param twoWindingsTransformer containing the PhaseTapChanger we want to convert
+     * @return The type of the UcteAngleRegulation
+     */
+    private UcteAngleRegulationType findRegulationType(TwoWindingsTransformer twoWindingsTransformer) {
+        if (isSymm(twoWindingsTransformer)) {
+            return UcteAngleRegulationType.SYMM;
+        } else {
+            return UcteAngleRegulationType.ASYM;
+        }
+    }
+
+    private boolean isSymm(TwoWindingsTransformer twoWindingsTransformer) {
+        for (int i = twoWindingsTransformer.getPhaseTapChanger().getLowTapPosition();
+             i < twoWindingsTransformer.getPhaseTapChanger().getHighTapPosition(); i++) {
+            if (twoWindingsTransformer.getPhaseTapChanger().getStep(i).getRho() != 1) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    UcteNodeCode createUcteNodeCode(String id, VoltageLevel voltageLevel, String country) {
```

Comment:
I used a String instead because we don't have an equivalent in iidm for a fictious country : a String allows me to pass things like `UcteCountryCode.XX.toString()` where I normally need to give a Country